examples/gno.land/p/gov/proposal/proposal.gno

```diff
@@ -16,11 +18,20 @@ func NewExecutor(callback func() error) Executor {
 	}
 }
 
+// NewCtxExecutor creates a new executor with the provided callback function.
+func NewCtxExecutor(callback func(ctx context.Context) error) Executor {
```

Comment:
Just curious, why would we use this as opposed to the regular callback?

Reply:
Nevermind, I found it

This is a way to not force custom executors on Realms, but to share an "approval token of govdao" for the call that the Realms can verify internally

Reply:
The two solutions are similar yet different, each with its own pros and cons. Neither is perfect enough to be the sole solution, so it makes sense to maintain both for now and eventually add new ones or remove old ones as needed.

One of the key differences is the potential for more complex multi-ownership flows. Instead of having only the creator's ownership, a third-party interceptor could verify the context and append its own ownership as well.

I explored this flow by proposing alternative initialization methods and considering how to manage proposals involving an asynchronous banker.